lib/puppetserver/ca/action/generate.rb

```diff
@@ -123,18 +122,16 @@ def generate_pki(settings, signing_digest, subject_alt_names = '')
           FileSystem.ensure_dir(settings[:privatekeydir])
           FileSystem.ensure_dir(settings[:publickeydir])
 
-          FileSystem.write_file(settings[:cert_inventory],
-                                inventory_entry(master_cert),
-                                0644)
-
           public_files = [
             [settings[:cacert], [int_cert, root_cert]],
             [settings[:cacrl], [int_crl, root_crl]],
             [settings[:hostcert], master_cert],
             [settings[:localcacert], [int_cert, root_cert]],
             [settings[:localcacrl], [int_crl, root_crl]],
-            [settings[:hostpubkey], [master_key.public_key]],
-            [settings[:capub], [int_key.public_key]],
+            [settings[:hostpubkey], master_key.public_key],
+            [settings[:capub], int_key.public_key],
+            [settings[:cert_inventory], inventory_entry(master_cert)],
+            [settings[:serial], "002"],
```

Comment:
nit: looking at the code we actually do the inventory serial number format here as well in the puppetserver code. We might want to do that here or as a separate ticket (kind of an unrelated bug fix).

Reply:
Do you mean we just shouldn't duplicate the logic? Or is there some formatting that is missing here? I think we might need it in both places because most signing will go through the API...

Reply:
I think the format for the serial number is the same in the serial file as it is in inventory file: https://github.com/puppetlabs/puppetserver/blob/master/src/clj/puppetlabs/puppetserver/certificate_authority.clj#L373

Reply:
Ah, gotcha. I'll fix that.